libs/src/api/file.js

```diff
@@ -147,6 +153,10 @@ class File extends Base {
   async uploadImage (file, square) {
     this.REQUIRES(Services.CREATOR_NODE)
     this.FILE_IS_VALID(file)
+
+    // Assign a creator_node_endpoint to the user if necessary
+    await this.User.assignReplicaSetIfNecessary()
```

Comment:
we call this on sign up before we even have a user created on the backend, would this work? it also seems kind of like a crutch to do it here. why did we choose uploadImage as the spot?

Reply:
when a user is uploading an image to a creator node, they are writing data to a content node. I think it's ideal to have a replica set store this image, and not the default UM. 

also, I added this here in the case that the sanity check fails. if that does happen, it would be ideal to have a followup where on a content node write to make sure the user has a rset